File: tests/fixtures/typescript/sample-repo/src/clock.ts
// Time-effect patterns.

export function nowMs(): number {
  // expect: Time=definite
  return Date.now();
}

export function timestamp(): Date {
  // expect: Time=definite
  return new Date();
}

export function highRes(): number {
  // expect: Time=definite
  return performance.now();
}

export function dateFromIso(iso: string): Date {
  // expect: (none — Pure)
  // `new Date(arg)` with an argument is pure construction.
  return new Date(iso);
}

export function nowLabel(): string {
  // expect: Time=possible
  return "now";
}

File: tests/fixtures/rust/sample-repo/src/clock.rs
// Time-effect patterns.

use std::thread;
use std::time::{Duration, Instant, SystemTime};

pub fn now_instant() -> Instant {
    // expect: Time=definite
    Instant::now()
}

pub fn now_system() -> SystemTime {
    // expect: Time=definite
    SystemTime::now()
}

pub fn pause(secs: u64) {
    // expect: Time=definite
    thread::sleep(Duration::from_secs(secs));
}

pub fn current_time_label() -> &'static str {
    // expect: Time=possible
    "now"
}
